src/fev/task.py

```diff
@@ -599,17 +599,16 @@ def _to_dataset(preds: datasets.Dataset | list[dict]) -> datasets.Dataset:
                 raise ValueError(f"predictions must be of type `datasets.Dataset` (received {type(preds)})")
             return preds
 
-        if self.is_multivariate:
-            if isinstance(predictions, datasets.DatasetDict):
-                pass
-            elif isinstance(predictions, dict):
-                predictions = datasets.DatasetDict({col: _to_dataset(preds) for col, preds in predictions.items()})
+        if not isinstance(predictions, datasets.DatasetDict):
+            if self.is_multivariate:
+                if isinstance(predictions, dict):
+                    predictions = datasets.DatasetDict({col: _to_dataset(preds) for col, preds in predictions.items()})
+                else:
+                    raise ValueError(
+                        f"predictions for multivariate tasks must be of type `datasets.DatasetDict` or `dict` (received {type(predictions)})"
+                    )
             else:
-                raise ValueError(
-                    f"predictions for multivariate tasks must be of type `datasets.DatasetDict` or `dict` (received {type(predictions)})"
-                )
-        else:
-            predictions = datasets.DatasetDict({self.target_column: _to_dataset(predictions)})
+                predictions = datasets.DatasetDict({self.target_column: _to_dataset(predictions)})
```

Comment:
Could you also add a new test case or extend an existing test in https://github.com/autogluon/fev/blob/main/test/test_task.py to ensure that  all prediction formats work with both univariate and multivariate tasks?